### DISCUSSION TITLE
Long run time for Seurat integration analysis on a dataset of about 10,000 cells

### DISCUSSION THREAD

**Discussion by chizuru0314** [satijalab/seurat#8083] [Q&A]
I'm currently running a Seurat integration analysis on a dataset consisting of approximately 10,000 cells on a server. However, the analysis has been running for five hours, and it has not completed yet. I'm concerned about the extended run time and would like to understand the possible reasons behind it.

Here is a summary of the steps I have taken:

1.Created a Seurat object and loaded the count data.
2.Performed data preprocessing steps such as normalization, identification of variable features, and scaling.
3.Ran the integration analysis using the CCAIntergration function with appropriate parameters.
```
brain <- IntegrateLayers(generic = brain@assays[["RNA"]],object = brain, 
                         method = CCAIntegration, orig.reduction = "pca", 
                         new.reduction = "integrated.cca",
                         verbose = FALSE)
```

4.The analysis has been running for five hours without completion.
I would appreciate any insights or suggestions regarding the following questions:

1.Is it expected for the integration analysis to take such a long time for a dataset with around 10,000 cells?
2.Are there any specific considerations or optimizations I should apply to speed up the analysis process?
3.Are there any potential factors or issues that could contribute to the prolonged run time?
4.Any guidance or recommendations would be greatly appreciated. Thank you for your assistance.

Please feel free to provide any additional information or ask for clarifications if needed.

**Comment by Dario-Rocha** [comment]
After I updated to the CRAN version of Seurat v5 and the Matrix package, many processess seem to have gotten orders of magnitude slower.
A script section that used to take about 20 minutes of runtime is now taking over 10 hours, on the same computer and without any modifications on the script itself.
